libr/asm/arch/avr/format.c

```diff
@@ -156,10 +156,34 @@ static int formatDisassembledOperand(avrDisassembleContext *context, char *strOp
 			fOptions.addressFieldWidth, context->longAddress);
 		break;
 	case OPERAND_IO_REGISTER:
-		retVal = sprintf(strOperand, "%s%02x",
-			OPERAND_PREFIX_IO_REGISTER,
-			dInstruction.operands[operandNum]);
+	{
+		char *current_register = NULL;
+		int is_register_found = 0;
+		switch(dInstruction.operands[operandNum])
+		{
+		    case 0x3f:
+		        current_register = "sreg\0";
+		        break;
+		    case 0x04:
+		        current_register = "adcl\0";
+		        break;
+		    case 0x05:
+		        current_register = "adch\0";
+		        break;
+		    default:
+		        is_register_found = -1;
+		        retVal = sprintf(strOperand, "0x%x", dInstruction.operands[operandNum]);
+		        break;
+		    break;
+		}
+
+		if (is_register_found == 0) {
```

Comment:
```suggestion
		if (is_register_found) {
```

Reply:
why use 0 for true and -1 for false, this condition is making the code less readable. please press the "commit suggestion" for all your changes. instead of closing and not applying them